### ISSUE TITLE
Delayed notifications 

### ISSUE THREAD

**Issue by mvevitsis** [Gedsh/InviZible#324]
I often have issues getting push notifications when using invizible. 

In my firewall I have whitelisted (allowed) the following 
UID 1000
UID 2000
Clat
DNS 
DNS tether
Internet connectivity check 
Internet time servers
Multicast DNS 
Network Manager 
Timezone Updater 
Google Play Services (for push)

Is there any way to diagnose this further? I'm on One UI.

**Comment by Gedsh** [comment]
What services do you use, Tor, DNSCrypt?

**Comment by mvevitsis** [comment]
> What services do you use, Tor, DNSCrypt?

Only dnscrypt

**Comment by Gedsh** [comment]
Do you use the DNSCrypt blacklists?

**Comment by mvevitsis** [comment]
> Do you use the DNSCrypt blacklists?

No, but I use controld. So I have lists, just they are remotely hosted. It would be helpful to know from the log when controld returns 0.0.0.0 specifically, not sure if this is something you could add for me.

**Comment by Gedsh** [comment]
> I have lists, just they are remotely hosted.

You may have blocked domains that are required for pushes to work properly.

> t would be helpful to know from the log when the DNS server returns 0.0.0.0 specifically

This has already been implemented. You will see the domain name in red color.

Also make sure you have battery and traffic optimization disabled for InviZible.

**Comment by mvevitsis** [comment]
> > I have lists, just they are remotely hosted.
> 
> You may have blocked domains that are required for pushes to work properly.
> 
> > t would be helpful to know from the log when the DNS server returns 0.0.0.0 specifically
> 
> This has already been implemented. You will see the domain name in red color.
> 
> Also make sure you have battery and traffic optimization disabled for InviZible.

How do I tell the difference between a request blocked by invizible (firewall) vs request blocked by the DNS server?

**Comment by Gedsh** [comment]
> How do I tell the difference between a request blocked by invizible (firewall) vs request blocked by the DNS server?

If the request is blocked by the firewall, you will see the name of the app. If the request is blocked by DNSCrypt, you will only see the domain. https://github.com/Gedsh/InviZible/wiki/Real-time-connections-log

**Comment by Gedsh** [comment]
> The issue seems to be resolved if I exclude Google Play Services from the socks5 proxy (which I have enabled for dnscrypt).

This means the problem is your controld server which is blocking something necessary for pushes to work.

> what DNS server are excluded apps (in this case only Google Play Services) using?

Excluded apps use the phone's native DNS resolver, such as your home router's DNS resolver, which generally is not a filtering resolver.

**Comment by mvevitsis** [comment]
> > The issue seems to be resolved if I exclude Google Play Services from the socks5 proxy (which I have enabled for dnscrypt).
> 
> This means the problem is your controld server which is blocking something necessary for pushes to work.
> 
> > what DNS server are excluded apps (in this case only Google Play Services) using?
> 
> Excluded apps use the phone's native DNS resolver, such as your home router's DNS resolver, which generally is not a filtering resolver.

It actually turned out to be a proxy/VPN issue and not a DNS issue - I resolved it by setting my wireguard VPN's keep alive to 25.

So you can rest assured there is no issue with invizible and play services.

**Comment by Gedsh** [comment]
> It actually turned out to be a proxy/VPN issue and not a DNS issue - I resolved it by setting my wireguard VPN's keepalive to 25s.

This is less likely, but also possible.

> you can rest assured there is no issue with invizible and play services.

So it seems the issue has been resolved.
